poet/sawtooth_poet/validator_registry/processor/handler.py

```diff
@@ -0,0 +1,334 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import logging
+import hashlib
+import base64
+import json
+
+from sawtooth_signing import pbct as signing
+
+from sawtooth_sdk.processor.state import StateEntry
+from sawtooth_sdk.client.future import FutureTimeoutError
+from sawtooth_sdk.processor.exceptions import InvalidTransaction
+from sawtooth_sdk.processor.exceptions import InternalError
+from sawtooth_sdk.protobuf.transaction_pb2 import TransactionHeader
+
+from sawtooth_poet.protobuf.validator_registry_pb2 import \
+    ValidatorInfo
+from sawtooth_poet.protobuf.validator_registry_pb2 import \
+    ValidatorMap
+from sawtooth_poet.protobuf.validator_registry_pb2 import \
+    ValidatorRegistryPayload
+
+LOGGER = logging.getLogger(__name__)
+
+STATE_TIMEOUT_SEC = 10
+
+
+def _get_address(key):
+    namespace = hashlib.sha256("validator_registry".encode()).hexdigest()[0:6]
```

Comment:
This can be a module-level constant - otherwise, you're computing this hash every `_get_address` call.